src/Log4netShipper/LogzioAppender.cs

```diff
@@ -38,14 +42,30 @@ private void WriteImpl(LoggingEvent loggingEvent)
 					{"thread", loggingEvent.ThreadName},
 					{"message", loggingEvent.RenderedMessage},
 					{"exception", loggingEvent.GetExceptionString()},
-					{"user", loggingEvent.UserName},
+					// log4net already adds this 'username' property which is added to values bellow
+					//{"user", loggingEvent.UserName},
+					{"processId", ProcessId}
 				};
 
 				foreach (var customField in _customFields)
 				{
 					values[customField.Key] = customField.Value;
 				}
 
+				var properties = loggingEvent.GetProperties();
+				foreach (DictionaryEntry property in properties)
+				{
+					var value = property.Value?.ToString();
+					if (string.IsNullOrWhiteSpace(value))
+						continue;
+
+					var key = property.Key.ToString();
+					key = key.Replace(":", "");
+					key = key.Replace(".", "");
+					key = key.Replace("log4net", "");
+					values["property" + key] = value;
```

Comment:
Hey, I did some major changes, including support for objects (non-string) in fields. You may grab the last revision if you want, or wait until I'll finish with the other improvements and release an official version. Thank you.